src/main/java/com/netflix/nebula/hollow/ApiGeneratorPlugin.java

```diff
@@ -31,13 +33,15 @@
      */
     @Override
     public void apply(Project project) {
-        Map<String, Object> taskPropertiesMap = new HashMap<>();
-        taskPropertiesMap.put("name", "generateHollowConsumerApi");
-        taskPropertiesMap.put("group", "hollow");
-        taskPropertiesMap.put("type", ApiGeneratorTask.class);
-        taskPropertiesMap.put("dependsOn", Collections.singletonList("build"));
-        project.getTasks().create(taskPropertiesMap);
-
-        project.getExtensions().create("hollow", ApiGeneratorExtension.class);
+        PluginContainer plugins = project.getPlugins();
+        if(plugins.hasPlugin(JavaPlugin.class)) {
```

Comment:
Decided to use `hasPlugin` since most of the times it gets applied way before custom plugins. 

While this could be replaced with `project.getPlugins().withType(JavaPlugin.class)`, I don't see it necessary for now. I think it's safe to fail if the java plugin gets applied after.